graph/connected_components_with_dsu.cpp

```diff
@@ -1,56 +1,110 @@
-#include <iostream>
-#include <set>
-#include <vector>
+/**
+ * @file
+ * @brief [Disjoint union](https://en.wikipedia.org/wiki/Disjoint_union)
+ *
+ * @details
+ * The Disjoint union is the technique to find connected component in graph efficiently.
+ *
+ * ### Algorithm
+ * In Graph, if you have to find out the number of connected components, there are 2 options
+ * 1. Depth first search
+ * 2. Disjoint union
+ * 1st option is inefficient, Disjoint union is the most optimal way to find this.
```

Comment:
```suggestion
 * 1st option is inefficient, Disjoint union is the most optimal way to find this.
 *
 * @author Unknown author
 * @author [Sagar Pandya](https://github.com/sagarpandyansit)
```

Reply:
@Panquesito7, can you tell me what is this change means? I could not understand.

Reply:
This will add the author name. Currently, I do not know who made this algorithm, so that's why I added `Unknown author`. Also added you because you significantly rewrite and improved the code. 🙂